src/core/adapters/recommendations.ts

```diff
@@ -1,7 +1,87 @@
+import * as effects from 'redux-saga/effects';
+import Configuration from '../configuration';
+import Store from '../store';
+import { fetch, sortBasedOn } from '../utils';
+import ConfigurationAdapter from './configuration';
+
 namespace Recommendations {
 
-  export const buildUrl = (customerId: string) =>
-    `https://${customerId}.groupbycloud.com/wisdom/v2/public/recommendations`;
+  export const buildUrl = (customerId: string, endpoint: string, mode: string) =>
+    `https://${customerId}.groupbycloud.com/wisdom/v2/public/recommendations/${endpoint}/_get${mode}`;
+
+  export const buildBody = (body: RecommendationsBody) => ({
+    method: 'POST',
+    body: JSON.stringify(body)
+  });
+
+  export const sortNavigations = ({ results, navigations }: Navigations): Navigation[] =>
+    sortBasedOn(results, navigations, (unsorted, sorted) => unsorted.name === sorted.name);
+
+  export const sortRefinements = ({ results, navigations }: Navigations): Navigation[] =>
+    navigations.reduce((resultsAcc, navigation) => {
+      const index = resultsAcc.findIndex(({ name }) => navigation.name === name);
+      if (index !== -1) {
+        resultsAcc = [...resultsAcc.slice(0, index), {
+          ...resultsAcc[index], refinements:
+            sortBasedOn(resultsAcc[index].refinements, navigation.values,
+              (unsorted: ValueRefinement, sorted) => unsorted.value.toLowerCase() === sorted.value.toLowerCase())
+         }, ...resultsAcc.slice(index + 1)];
+      }
+      return resultsAcc;
+    }, results);
+
+  export const pinNavigations = ({ results, config }: NavigationsAndConfig): Navigation[] => {
+    const pinnedArray = ConfigurationAdapter.extractNavigationsPinned(config);
+    return sortBasedOn(results, pinnedArray, (unsorted, pinnedName) => unsorted.name === pinnedName);
+  };
+
+  export const pinRefinements = ({ results, config }: NavigationsAndConfig): Navigation[] => {
+    const newNavigations = [];
+    const pinnedRefinements: Configuration.Recommendations.Pinned =
+      ConfigurationAdapter.extractRefinementsPinned(config);
+    const pinnedRefinementsNavigationsArray: Store.Recommendations.Navigation[] =
+      Object.keys(pinnedRefinements).map((key) =>
+        ({
+          name: key,
+          values: pinnedRefinements[key].map((value) => ({
+            value,
+            count: -1
+          }))
+        }));
+    return sortRefinements({ results, navigations: pinnedRefinementsNavigationsArray });
+  };
+
+  // tslint:disable-next-line max-line-length
+  export const sortAndPinNavigations = (availableNavigations: Navigation[], navigations: Store.Recommendations.Navigation[], config: Configuration): Navigation[] => {
+    const iNav = ConfigurationAdapter.extractINav(config);
+    const noop = ((x) => x.results);
+    const transformations = [
+      iNav.navigations.sort ? sortNavigations : noop,
+      Array.isArray(iNav.navigations.pinned) ? pinNavigations : noop,
```

Comment:
Why are you checking isArray here? Does pinned have the option to be something else in the configuration?

Reply:
I just figured I'd to a tiny bit more sanitization on top of checking if it exists, since for non-array pinned that'd error out pinNavigations. I can just check if it's truthy if that's preferable.